### PR TITLE
update measure parameter endpoint 

### PR DESCRIPTION
**Pull request by deekim**
### primary
_update_ `/fhir/measure/{measure_id}/parameters` endpoint
to return a `Response` object that contains parameter information from the `Measure` model (bcda4a5), 
previously expected to be on the `Library` (a0326d6)

#### measure parameter default
- pull parameter default values from `ParameterDefinition` (906efde)
→ I misunderstood the IG guidance and assumed it was directly on the `Measure` (#69)

#### exception handling
- return ~`404`~`500` when requested `measure` was not found (df4e51e)
- return exception details for `500` error codes (5628223)

---

## references

- [WFFHCOHORT-278](https://jsw.ibm.com/browse/WFFHCOHORT-278)
- [CDM Library StructureDefinition](https://pages.github.ibm.com/watson-health-fhir-server/ig-common-data-model/StructureDefinition-cdm-library.html)
- [MeasureParameter extension example](https://github.ibm.com/watson-health-fhir-server/ig-common-data-model/blob/4750f53d14e1bdec7902fe4942023b806a732777/data/resources/quality-measure/QualityMeasure_Instance.fsh#L88-L91)

```
* extension[measureParameter][1].valueParameterDefinition.name = #complianceDisplayName
* extension[measureParameter][1].valueParameterDefinition.use = #out 
* extension[measureParameter][1].valueParameterDefinition.type = #string
* extension[measureParameter][1].valueParameterDefinition.extension[defaultValue].valueString = "Pharyngitis Testing"
```

---

## requested changes
- [x] handle `null` on `getParametersForMeasureIdentifier`

### CODE REVIEW

File: cohort-engine/src/main/java/com/ibm/cohort/engine/measure/MeasureEvaluator.java [hunk 1]
```diff
@@ -35,7 +37,8 @@
  * server.
  */
 public class MeasureEvaluator {
-	protected static final String PARAMETER_EXTENSION_URL = "http://hl7.org/fhir/us/cqfmeasures/StructureDefinition/cqfm-parameter";
+	protected static final String MEASURE_PARAMETER_URL = "http://ibm.com/fhir/cdm/StructureDefinition/measure-parameter";
+	protected static final String PARAMETER_DEFAULT_URL = "http://ibm.com/fhir/cdm/StructureDefinition/default-value";
```
Comment (deekim) [inline cohort-engine/src/main/java/com/ibm/cohort/engine/measure/MeasureEvaluator.java:41]: Ideally, I would move these to `CDMConstants.java` once #74 is merged back

**Review by shelleyMurphy — APPROVED** [review]
Kept getting distracted, but this looks sensible to me.

File: cohort-engine-api/src/main/java/com/ibm/cohort/engine/api/service/FHIRRestUtils.java [hunk 2]
```diff
@@ -153,7 +153,7 @@ public static IGenericClient getFHIRClient(String fhirEndpoint, String bearerTok
 		RestFhirMeasureResolutionProvider msp = new RestFhirMeasureResolutionProvider(measureClient);
 		Measure measure = msp.resolveMeasureByIdentifier(identifier, measureVersion);
```
Comment (tabishop) [inline cohort-engine-api/src/main/java/com/ibm/cohort/engine/api/service/FHIRRestUtils.java:155]: Think we need a null check here on the measure as well, no?

**Review by tabishop — CHANGES_REQUESTED** [review]
Think we are missing a null check/exception throw.

File: cohort-engine-api/src/main/java/com/ibm/cohort/engine/api/service/FHIRRestUtils.java (hunk 2 above)
Comment (deekim) [reply cohort-engine-api/src/main/java/com/ibm/cohort/engine/api/service/FHIRRestUtils.java:155]: Yep, good call.
I forgot there were two codepaths for getting measures.

**Review by tabishop — APPROVED** [review]
looks good

File: cohort-engine/src/main/java/com/ibm/cohort/engine/measure/MeasureEvaluator.java [hunk 3]
```diff
@@ -205,13 +208,27 @@ public MeasureReport evaluatePatientMeasure(Measure measure, String patientId, S
 		return evaluation.evaluatePatientMeasure(measure, seed.getContext(), patientId);
 	}
 
+	private void setDefaultValue(Context context, ParameterDefinition parameterDefinition) {
+		parameterDefinition.getExtension().stream()
+				.filter(MeasureEvaluator::isDefaultValue)
+				.forEach(defaultValue ->
+						         context.setParameter(
+						         		null,
```
Comment (msargentibm) [inline cohort-engine/src/main/java/com/ibm/cohort/engine/measure/MeasureEvaluator.java:216]: I know this is just refactored code as part of this PR, but I wanted to confirm that `null` here makes sure that the parameter is global/not tied to a specific library.

**Review by msargentibm — APPROVED** [review]
Only comment is a question for one line that was part of the previous PR for parameters.

**Review by jaigoradia — APPROVED** [review]
Approved.

**Review by dkwasny-ibm — APPROVED** [review]

**Pull request merged**